src/container/TreeContainer/Base/index.ts

```diff
@@ -241,26 +200,41 @@ abstract class TreeContainer<K, V> extends Container<K | [K, V]> {
       this._header._right = swapNode._parent;
     }
     this._eraseNodeSelfBalance(swapNode);
-    const _parent = swapNode._parent!;
+    let _parent = swapNode._parent as TreeNodeEnableIndex<K, V>;
     if (swapNode === _parent._left) {
       _parent._left = undefined;
     } else _parent._right = undefined;
     this._length -= 1;
     this._root!._color = TreeNodeColor.BLACK;
-    return _parent;
+    if (this.enableIndex) {
+      while (_parent !== this._header) {
+        _parent._subTreeSize -= 1;
+        _parent = _parent._parent as TreeNodeEnableIndex<K, V>;
+      }
+    }
   }
   /**
    * @internal
    */
-  protected _inOrderTraversal(
-    curNode: TreeNode<K, V> | undefined,
-    callback: (curNode: TreeNode<K, V>) => boolean
-  ): boolean {
-    if (curNode === undefined) return false;
-    const ifReturn = this._inOrderTraversal(curNode._left, callback);
-    if (ifReturn) return true;
-    if (callback(curNode)) return true;
-    return this._inOrderTraversal(curNode._right, callback);
+  protected _inOrderTraversal(pos: number) {
+    pos += 1;
+    const nodeList = [];
```

Comment:
Generating a list of items for each inOrderTraversal seems quite not GC-friendly.
It is fatal in applications that perform every frame foreach, such as games.

Reply:
Can you give a example or more information about the downside of `Generating a list of items`?

Reply:
The return type of the `_inOrderTraversal` method is array
after generating a `nodeList`, (line 221)
the item is pushed back to the `nodeList` through inorder traversal. (line 230)

Reply:
Is the reason for this implementation because callback can cause side effects?

In previous implementations, the function received callback, but why did it replaced to array?

Reply:
https://softwareengineering.stackexchange.com/questions/179863/performance-recursion-vs-iteration-in-javascript

Reply:
It is certainly performance-friendly to change recursion to iteration.
However, it does not seem necessary to create an array just because it is implemented in the form of iteration.

Reply:
This is always used for `forEach` and `Symbol.iterator` to do iterate.

Reply:
Here's a quick benchmark result.
Obviously, non-recursive implementations is about 5 times faster, but you can see that they consume about 30 times more memory.

code:
```typescript
const orderedSet = new OrderedSet<number>();

for (let i = 0; i < 50000000; i++) {
    orderedSet.insert(i);
}

const startHeapUsed = process.memoryUsage().heapUsed;
const startTimestamp = Date.now();
console.log(`startHeapUsed: ${startHeapUsed}`);
orderedSet.forEach((key, value) => {});
const endHeapUsed = process.memoryUsage().heapUsed;
console.log(`endHeapUsed: ${endHeapUsed}`);

console.log(`mem diff: ${endHeapUsed - startHeapUsed}`);
console.log(`time: ${Date.now() - startTimestamp}`);
```

on 4.4.0
```besh
startHeapUsed: 3604797016
endHeapUsed: 4005968360
mem diff: 401171344
time: 4353
```

on 4.3.0
```besh
startHeapUsed: 3604815208
endHeapUsed: 3618398288
mem diff: 13583080
time: 23392
```

Reply:
Can this PR solve your problem? https://github.com/js-sdsl/js-sdsl/pull/166